manual/root_io/index.md

```diff
@@ -1,348 +1,222 @@
-
-	- `Isomething`: Adding an include path, so that `rootcling` can find the files included in `Header1.h`, `Header2.h`, etc.
-
-    - `DSOMETHING`: Define a preprocessor macro, which is sometimes needed to parse the header files.
-
-- `Header1.h Header2.h...`: The headers files.
-
-- `Linkdef.h`: Tells `rootcling`, which classes should be added to the dictionary, → see [Selecting dictionary entries: Linkdef.h](#selecting-dictionary-entries-linkdefh).
+In order to store your C++ types, ROOT needs to know about its data members and their types.
+ROOT can extract that information from your header files with the help of its C++ interpreter, Cling, and store it in **ROOT dictionaries**.
+A dictionary ("reflection database") contains information about the types and functions that are available in a library in the form of C++ code that can be linked into your application.
+These dictionaries can be generated automatically by ROOT in a few different ways, which we describe in the following section.
 
 > **Note**
 >
-> Dictionaries that are used within the same project must have unique names.
->
-> Compiled object files relative to dictionary source files cannot reside in the same library or in two libraries loaded by the same application if the original source files have the same name.
-_**Example**_
-
-In the first step, a `TEvent` and a `TTrack` class is defined. Next an event object is created to add tracks to it. The track objects have a pointer to their event. This shows that the I/O system correctly handles circular references.
+> Dictionaries are _not_ required to use a given C++ type in the ROOT interpreter (e.g. in the ROOT prompt, via PyROOT or in Jupyter notebooks). They are only required to perform I/O of user-defined classes.
 
-In the second step, a `TEvent` and a `TTrack` call are implemented.<br/>After that you can use `rootcling` to manually generate a directory. This generates the `eventdict.cxx` file.
 
-**The TEvent.h header**
+## Generating dictionaries
 
-{% highlight C++ %}
-#ifndef __TEvent__
-#define __TEvent__
-#include "TObject.h"
-class TCollection;
-class TTrack;
-
-class TEvent : public TObject {
-private:
-   Int_t fId; // Event sequential id
-   Float_t fTotalMom;       // Total momentum.
-   TCollection *fTracks;    // Collection of tracks.
-public:
-   TEvent() { fId = 0; fTracks = 0; }
-   TEvent(Int_t id);
-   ~TEvent();
-   void AddTrack(TTrack *t);
-   Int_t GetId() const { return fId; }
-   Int_t GetNoTracks() const;
-   void Print(Option_t *opt="");
-   Float_t TotalMomentum();
-   ClassDef(TEvent,1);     //Simple event class.
-};
-{% endhighlight %}
+A dictionary consists of a C++ source file, which contains the type information needed by Cling and ROOT's I/O subsystem. This source file needs to be generated from the library's headers and then compiled and linked to the application that needs to perform I/O of the included classes.
 
-**The TTrack.h header**
+There are three ways to generate a dictionary:
```

Comment:
Does it matter to the intended reader that 2 of our the 3 methods turns around and invoke/use the 3rd?  [Also I noting that we are not mentioning genreflex but that is probably a good thing)

Reply:
in the explanation of cmake and aclic we mention that they invoke rootcling under the hood

Reply:
Fair enough.